gateway/ln-gateway/src/rpc/rpc_server.rs

```diff
@@ -155,6 +154,7 @@ fn v1_routes(gateway: Arc<Gateway>) -> Router {
             CREATE_BOLT11_INVOICE_FOR_SELF_ENDPOINT,
             post(create_invoice_for_self),
         )
+        .route(PAY_INVOICE_SELF_ENDPOINT, post(pay_invoice_self))
```

Comment:
This needs to be authenticated, otherwise anyone can spend the gateway's funds.

`spend_ecash` and `receive_ecash` should also be authenticated

Reply:
Yikes, good catch. Fixed